src/DependencyInjection/Compiler/AdminExtensionCompilerPass.php

```diff
@@ -32,7 +32,11 @@ public function process(ContainerBuilder $container): void
 
         foreach ($container->findTaggedServiceIds('sonata.admin') as $id => $attributes) {
             $admin = $container->getDefinition($id);
-            $modelClass = $container->getParameterBag()->resolveValue($admin->getArgument(1));
+
+            // NEXT_MAJOR: Remove the $admin->getArguments()[1] fallback.
+            $modelClassName = $attributes[0]['model_class'] ?? $admin->getArguments()[1] ?? null;
```

Comment:
Did you try this ? I think it should be
```suggestion
            $modelClassName = $attributes['model_class'] ?? $admin->getArguments()[1] ?? null;
```
instead, no ?

Like : https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/DependencyInjection/Compiler/ExtensionCompilerPass.php#L146

Reply:
I thought it was strange too,  but $attributes contains an array.
$attributes = array(
   0 => array( 'model_class' => 'App\Entity\User' , 'manager_type' => 'orm')
)

Reply:
Looking at https://github.com/sonata-project/SonataAdminBundle/blob/4.x/src/DependencyInjection/Compiler/ExtensionCompilerPass.php#L55 I understand, the name attributes is wrong. `tags` is better.

Also, it's better to use a foreach on the attributes. Like this then.
```
foreach ($container->findTaggedServiceIds('sonata.admin') as $id => $tags) {
            $admin = $container->getDefinition($id);

            // NEXT_MAJOR: Remove this line.
            $defaultModelClass = $admin->getArguments()[1] ?? null;
            foreach ($tags as $attributes) {
                // NEXT_MAJOR: Remove the fallback to $defaultModelClass and use null instead.
                $modelClass = $attributes['model_class'] ?? $defaultModelClass;
                if (null === $modelClass) {
                    throw new InvalidArgumentException(sprintf('Missing tag attribute "model_class" on service "%s".', $id));
                }

                $class = $container->getParameterBag()->resolveValue($modelClass);
                if (!\is_string($class)) {
                    throw new \TypeError(sprintf(
                        'Tag attribute "model_class" for service "%s" must be of type string, %s given.',
                        $id,
                        \is_object($class) ? \get_class($class) : \gettype($class)
                    ));
                }

                if (!class_exists($class)) {
                    continue;
                }

                $modelClassReflection = new \ReflectionClass($modelClass);
                foreach ($adminExtensionReferences as $type => $reference) {
                    foreach ($translationTargets[$type]['implements'] as $interface) {
                        if ($modelClassReflection->implementsInterface($interface)) {
                            $admin->addMethodCall('addExtension', [$reference]);
                        }
                    }
                    foreach ($translationTargets[$type]['instanceof'] as $class) {
                        if ($modelClassReflection->getName() === $class || $modelClassReflection->isSubclassOf($class)) {
                            $admin->addMethodCall('addExtension', [$reference]);
                        }
                    }
                }
            }
}
```

Reply:
@VincentLanglet 
- renamed $attributes to $tags
- added foreach for looping the tags